### PR TITLE
[DOPS-2942]: Update substrate build

### PR DESCRIPTION
**Pull request by f33r0**
# Task

[DOPS-2942]: Update substrate build

## Changes

1. Updated env docker image (fixed build and coverage there)
2. Updated build workflow. Tests do not runs twice.
3. Clippy made optional (Works on gh actions)
4. Move coverage report to sonar https://github.com/sora-xor/sora2-network/pull/903 (by @C4tWithShell),  **but we can publish reports in jenkins if necessary, the problem is that these coverage reports load jenkins itself** This is the way to convert lcov report to cobertura report. Because the direct cobertura report is 1.5 GB in size. Fixed coverage and test step.
5. Updated subwasm installation method and updated subwasm itself

jenkins-library PR:
https://github.com/soramitsu/jenkins-library/pull/567

## Author

Signed-off-by: Nikita Zaporozhets <zaporozhets@soramitsu.co.jp>

[DOPS-2942]: https://soramitsu.atlassian.net/browse/DOPS-2942

### CODE REVIEW

File: housekeeping/docker/env/Dockerfile [hunk 1]
```diff
@@ -1,59 +1,46 @@
...
+ENV CCLANG_VER=14
+ARG DEBIAN_FRONTEND=noninteractive
 
-# Install dependencies
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
     ca-certificates apt-transport-https gnupg \
     libssl-dev pkg-config \
     curl \
+    wget \
```
Comment (Cre-eD) [inline housekeeping/docker/env/Dockerfile:22]: why do you need both?

**Review by safinsaf — APPROVED** [review]

**Review by RamilMus — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 
 def pipeline = new org.rust.AppPipeline(steps: this,
       initSubmodules: true,
-      envImageName: 'docker.soramitsu.co.jp/sora2/env:sub4',
+      envImageName: 'docker.soramitsu.co.jp/sora2/env:env',
       appImageName: 'docker.soramitsu.co.jp/sora2/substrate',
       codeCoverageCommand: './housekeeping/coverage.sh',
       cargoDoc: true,
       smartContractScanner: false,
+      clippyLinter: false,
```
Comment (Cre-eD) [inline Jenkinsfile:10]: why?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
       cargoClippyTag: ':substrate',
       cargoClippyCmds: ['housekeeping/clippy.sh'],
-      buildTestCmds: ['housekeeping/build.sh'],
+      buildTestCmds: 'housekeeping/build.sh',
```
Comment (Cre-eD) [inline Jenkinsfile:13]: did you remove the list?

File: Jenkinsfile-env [hunk 4]
```diff
@@ -4,9 +4,9 @@ def pipeline = new org.docker.AppPipeline(steps: this,
...
     dockerRegistryCred:           'bot-sora2-rw',
     dockerFileName:               "housekeeping/docker/env/Dockerfile",
-    triggerCommonBuildExpression: (env.BRANCH_NAME in ['substrate-4.0.0']),
+    triggerCommonBuildExpression: (env.BRANCH_NAME in ['develop', 'master']),
     triggerCommonBuildGrepRegex:  'housekeeping',
-    secretScannerExclusion:       '.*Cargo.toml',
+    secretScannerExclusion:       '.*Cargo.toml$|.*env\$',
     nameCI:                       'env-CI',
-    dockerImageTags:              ['substrate-4.0.0':'sub4'])
+    dockerImageTags:              ['master': 'env', 'develop': 'env'])
```
Comment (Cre-eD) [inline Jenkinsfile-env:11]: what we have as a default?

File: bridge-docker/.env [hunk 5]
```diff
@@ -0,0 +1,3 @@
+DATABASE_URL="postgres://postgres:${DATABASE_PASSWORD}@bridge-postgres:5432/postgres?sslmode=disable"
+DATABASE_PASSWORD="postgres"
+SECRET_KEY_BASE="56NtB48ear7+wMSf0IQuWDAAazhpb31qyc7GiyspBP2vh7t5zlCsF5QDv76chXeN"
```
Comment (Cre-eD) [inline bridge-docker/.env:3]: what is this key for?

Comment (Cre-eD) [reply bridge-docker/.env:3]: If you have it in the docker-compose, why is it duplicated here?

**Review by Cre-eD — APPROVED** [review]

**Pull request merged**